### PR TITLE
Add alert blocks in markdown

### LINKED ISSUE #28316: Update Note and Warning markdown highlighting with GitHub's changes
### Feature Description

Since July 21st, GitHub has made a few changes to their feature. We introduced it into Gitea with https://github.com/go-gitea/gitea/pull/21711. Now we need to update the implementation to match GitHub's, including a syntax change. An example of the current implementation on GitHub:

> [!NOTE]  
> Highlights information that users should take into account, even when skimming.

> [!TIP]
> Optional information to help a user be more successful.

> [!IMPORTANT]  
> Crucial information necessary for users to succeed.

> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.

> [!CAUTION]
> Negative potential consequences of an action.

Docs: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts https://github.com/orgs/community/discussions/16925

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by yardenshoham**
- Follows https://github.com/go-gitea/gitea/pull/21711
- Closes https://github.com/go-gitea/gitea/issues/28316

Implement GitHub's alert blocks markdown feature

Docs:
- https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts
- https://github.com/orgs/community/discussions/16925

### Before
image

### After
image

## :warning: BREAKING :warning:

The old syntax no longer works

How to migrate:

If you used
```md
> **Note** My note
```

Switch to
```md
> [!NOTE]
> My note
```

### CODE REVIEW

**Comment by lafriks** [comment]
Would be nice to also fix the title style as the icon looks really out of order

**Comment by yardenshoham** [comment]
Yeah I don't know CSS it's black magic

**Comment by denyskon** [comment]
I will take look at the css tomorrow 😉

**Comment by silverwind** [comment]
"Tip" icon not colored correctly in the screenshot.

**Comment by yardenshoham** [comment]
Updated the icon vertical-align

# Before

image

# After

image

**Comment by yardenshoham** [comment]
Fixed tip color

# Before

image

# After

image

File: modules/markup/markdown/goldmark.go [hunk 1]
```diff
@@ -197,18 +196,62 @@ func (g *ASTTransformer) Transform(node *ast.Document, reader text.Reader, pc pa
...
+			// Text: "!TYPE"
+			// Text(SoftLineBreak): "]"
+
+			// grab these nodes
+			firstParagraph := v.FirstChild()
+			if firstParagraph.ChildCount() < 3 {
+				return ast.WalkContinue, nil
+			}
+			firstTextNode, ok := firstParagraph.FirstChild().(*ast.Text)
+			if !ok {
```
Comment (lafriks) [inline modules/markup/markdown/goldmark.go]: Fail fast
```suggestion
			if !ok || string(firstTextNode.Segment.Value(reader.Source())) != "[" {
```

same for second and third node

Comment (yardenshoham) [reply modules/markup/markdown/goldmark.go]: I did it the way it is right now for readability. Do you prefer it this way?

Comment (yardenshoham) [reply modules/markup/markdown/goldmark.go]: Ah yes it makes sense to fail fast https://github.com/go-gitea/gitea/pull/29121/commits/03882cf0864d3a09d74b205e11f93e3fb64a90f1

**Comment by yardenshoham** [comment]
Oof these look bad in light theme, I need better colors
image

**Comment by yardenshoham** [comment]
Fixed
image

**Comment by yardenshoham** [comment]
Added some margin-right to the icon and dropped the boldness one level

# Before
image

# After
image

**Comment by silverwind** [comment]
Contrast on dark theme seems a bit low for Note, Caution and Tip. Do we have better suitable variables (1 level lighter?).

**Comment by yardenshoham** [comment]
I changed Note and Caution. I think Tip will look bad on light theme if I change it
# Before
image

# After
image

**Comment by silverwind** [comment]
Pushed a tweak to purple, looks good enough for me on both themes:

Screenshot 2024-02-10 at 18 10 33
Screenshot 2024-02-10 at 18 10 53

**Comment by silverwind** [comment]
The problem with the tip icon is strange, I will investigate.

**Comment by silverwind** [comment]
@yardenshoham the problem is the tip icon is missing the `class` attribute and only `.svg` inherits the color from CSS `color`. I reverted the CSS hacks, can you check where the class goes missing in backend? It is present on other icons, for example `class="svg octicon-alert"`.

**Comment by yardenshoham** [comment]
Fixed

**Review by silverwind — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

File: modules/markup/markdown/goldmark.go [hunk 2]
```diff
@@ -197,18 +196,55 @@ func (g *ASTTransformer) Transform(node *ast.Document, reader text.Reader, pc pa
...
-				}
+		case *ast.Blockquote:
+			// We only want attention blockquotes when the AST looks like:
+			// Text: "["
+			// Text: "!TYPE"
+			// Text(SoftLineBreak): "]"
+
+			// grab these nodes and make sure we adhere to the attention blockquote structure
+			firstParagraph := v.FirstChild()
+			if firstParagraph.ChildCount() < 3 {
```
Comment (wxiaoguang) [inline modules/markup/markdown/goldmark.go:207]: The `firstParagraph` might be `nil`, need to fix a panic:  Fix markdown render #33870

**Pull request merged**